apps/condo/domains/meter/access/MeterReading.js

```diff
@@ -10,31 +12,38 @@ const { queryOrganizationEmployeeFromRelatedOrganizationFor, queryOrganizationEm
 const { Resident: ResidentServerUtils } = require('@condo/domains/resident/utils/serverSchema')
 const { get } = require('lodash')
 
-
 async function canReadMeterReadings ({ authentication: { item: user }, context }) {
     if (!user) return throwAuthenticationError()
     if (user.isAdmin || user.isSupport) return {}
 
     const userId = user.id
     if (user.type === RESIDENT) {
-        const residents = await ResidentServerUtils.getAll(context, { user: { id: userId } })
-
-        for (const resident of residents) {
-            const residentPropertyId = get(resident, ['property', 'id'])
-            const residentUnitName = get(resident, 'unitName')
-
-            const serviceConsumers = await ServiceConsumer.getAll(context, {
-                resident: { id: resident.id },
-            })
-            const serviceConsumerAccounts = serviceConsumers.map(serviceConsumer => serviceConsumer.accountNumber)
-
-            const meters = await Meter.getAll(context, {
-                property: { id: residentPropertyId },
-                unitName: residentUnitName,
+        const { meter: meterQuery } = getMetersDataFromRequestBody(context)
+        const [meter] = await Meter.getAll(context, {
+            ...meterQuery,
+        })
+        const propertyId = get(meter, ['property', 'id'], null)
+        const unitName = get(meter, 'unitName', null)
+        const [resident] = await ResidentServerUtils.getAll(context, {
+            user: { id: userId },
+            property: { id: propertyId },
+            unitName,
+            deletedAt: null,
+        })
+        const residentId = get(resident, 'id', null)
+        const serviceConsumers = await ServiceConsumer.getAll(context, {
+            resident: { id: residentId },
+            deletedAt: null,
+        })
+        const serviceConsumerAccounts = serviceConsumers.map(serviceConsumer => serviceConsumer.accountNumber)
+
+        return {
+            meter: {
+                ...meter,
+                property: { id: propertyId },
```

Comment:
What if property will be deleted?

Reply:
now there is a MeterReading on the deleted property, added deletedAt